File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-11.rs
impl Solution {
    pub fn length_of_longest_substring(s: String) -> i32 {
        let mut l:usize=0;
        let mut r:usize=0;
        let s_vec:Vec<char>=s.chars().collect();
        let len:usize=s_vec.len();
        let mut max:i32=0;
        let mut set:HashSet<char>=HashSet::new();
        while r<len{
            while r<len && !set.contains(&s_vec[r]){
                set.insert(s_vec[r]);
                r+=1;
            }
            max=max.max((r-l) as i32);
            set.remove(&s_vec[l]);
            l+=1;
        }
        return max;
    }
}

File: Data Structures & Algorithms/max-water-container/submission-11.java
class Solution {
    public int maxArea(int[] heights) {
        int l=0;
        int r=heights.length-1;
        int max=0;
        while(l<r){
            int l_h=heights[l];
            int r_h=heights[r];
            int h=0;
            if(l_h<r_h){
                l++;
                h=l_h;
            }else{
                r--;
                h=r_h;
            }
            max=Math.max(max, (r-l+1)*h);
        }
        return max;
    }
}
